docs/github_pages_version_file.py

```diff
@@ -0,0 +1,175 @@
+        )
+    # Otherwise see if we should return something from the process
+    if return_stdout:
+        return process.stdout.decode("utf8")
+    elif return_exit_status:
+        return process.returncode
+    else:
+        return None
+
+
+def get_sorted_tags_list() -> List[str]:
+    """Get a list of sorted tags in descending order from the repository."""
+    stdout = run_command("git tag -l --sort=-v:refname", return_stdout=True)
+    assert isinstance(stdout, str), "Error getting tags"
+    return stdout.strip().split("\n")
+
+
+def get_branch_contents(remote: str, branch: str) -> List[str]:
+    """Get the list of directories in a branch."""
+    stdout = run_command(
+        f"git ls-tree -d --name-only {remote}/{branch}", return_stdout=True
+    )
+    assert isinstance(
+        stdout, str
+    ), f"Error getting directory contents of {remote}/{branch}"
+    return stdout.strip().split("\n")
+
+
+def get_docs_url(github_org: str, project: str, version: str):
+    """Get the URL for a specific build of the project."""
+    return f"https://{github_org}.github.io/{project}/{version}"
+
+
+def get_version_file_string(
+    main_branches: List[GitHubPagesBuild],
+    branches: List[GitHubPagesBuild],
+    releases: List[GitHubPagesBuild],
+) -> str:
+    """Create the string for the RST file."""
+    file_str = ""
+    # Add branches first
+    for build in main_branches:
+        file_str += f"{build.version} <{build.url}>\n"
+    for build in branches:
+        file_str += f"{build.version} <{build.url}>\n"
+    # Add releases
+    for build in releases:
+        file_str += f"{build.version} <{build.url}>\n"
+    return file_str
+
+
+def push_file_to_GitHub(file_string: str, filename: str, remote: str, branch: str):
+    """Push the version file to the GitHub Pages branch."""
+    # Check out a local branch tracking the GitHub Pages branch
+    run_command(f"git checkout -tq {remote}/{branch}")
+
+    try:
+        # Write the file
+        with open(filename, "w") as output_file:
+            output_file.write(file_string)
+
+        # Check if file is already tracked
+        status = run_command(
+            f"git ls-files --error-unmatch {filename}",
+            return_exit_status=True,
+            fail_on_bad_status=False,
+        )
+
+        # Status of zero means tracked. So check if file has changed
+        if not status:
+            status = run_command(
+                f"git diff --exit-code {filename}", return_exit_status=True
+            )
+
+        # If status is one (either untracked or file has changed) then commit the file
+        if status:
+            print(f"File {filename} being commited to {remote}/{branch}.")
+            # Add the file
+            run_command(f"git add {filename}")
+
+            # Commit the file
+            run_command("git commit -qm Update-version-file")
+
+            # Push to remote
+            run_command("git push -q")
```

Comment:
We have permissions to do this? That makes things easier!

Reply:
Could we do the deploy of the build/html directory too then? Save our use of the external action:
https://github.com/dls-controls/scanspec/blob/test/gh-pages/.github/workflows/docs.yml#L41-L60

Reply:
Looks like it! I just ran this build which commits the file: https://github.com/dls-controls/scanspec/runs/4379379099?check_suite_focus=true

The file was committed to the test branch here: https://github.com/dls-controls/scanspec/blob/test/gh-pages-file/version.txt